packages/runtime/garbage-collector/src/utils.ts

```diff
@@ -179,10 +179,17 @@ export function concatGarbageCollectionData(gcData1: IGarbageCollectionData, gcD
 }
 
 export class GCDataBuilder implements IGarbageCollectionData {
-    public readonly gcNodes: { [ id: string ]: string[] } = {};
+    private readonly gcNodesSet: { [ id: string ]: Set<string> } = {};
+    public get gcNodes(): { [ id: string ]: string[] } {
+        const gcNodes = {};
+        for (const [nodeId, outboundRoutes] of Object.entries(this.gcNodesSet)) {
+            gcNodes[nodeId] = [...outboundRoutes];
+        }
+        return gcNodes;
+    }
 
     public addNode(id: string, outboundRoutes: string[]) {
-        this.gcNodes[id] = Array.from(outboundRoutes);
+        this.gcNodesSet[id] = new Set(outboundRoutes);
```

Comment:
`unpackChildNodesGCDetails` should also be updated to form a Set from the outboundRoutes. That is the function that is used when sending the GC data down from container runtime to DDSs. That way you don't have to individually convert them to set at each layer.
It will also remove all duplicate routes across container runtime -> data store boundary.